### ISSUE TITLE
don't see any devices

### ISSUE THREAD

**Issue by dnyt84** [cconard96/jamf#58]
Hello,

we use Jamf Cloud.
I configured the Plugin via Plugins->Jamf Plugin->Configure plugin

Jss Server: https://oursubdomain.jamfcloud.com/
JSS User
JSS Password

Everything is set to yes except for Auto Import
Sync Interval 120m

Default Settings
Default Status In use

I can't seem to get any devices to show up.
Plugins->Jamf Plugin->Import devices I click on Import or Discover now
Plugins->Jamf Plugin->Merge existing devices 
All pages are empty, i executed the cron 
Setup-Automatic actions -> importJamf
I changed that cron to CLI instead of GLPI

My username has admin privileges 

I can't find any logs that indicate why this is not working.
Any help is appreciated!

Daniel

**Comment by cconard96** [comment]
If there is an API error, there should be an entry in files/_log/php-errors.log. If there is a DB issue, there should be an entry in files/_log/sql-errors.log.

Could I also get the GLPI and plugin versions so I can try to replicate the issue here?

**Comment by dnyt84** [comment]
Hello,

I am running glpi 9.5.1 and master download for Jamf Plugin.
PHP 7.2.32 - Apache/2.4.6 (CentOS)

When I execute the cron I get:
2020-08-07 00:19:39 [2@glpi001.xxx]
Undefined function PluginJamfCron::cronimportJamf (for cron)

When I click on Discover and Import I get: 
2020-08-07 00:22:41 [2@glpi001.den01.pop]
Undefined function PluginJamfCron::cronimportJamf (for cron)
[2020-08-07 00:22:47] glpiphplog.CRITICAL:   *** Uncaught Exception RuntimeException: Required argument missing! in /var/www/glpi/plugins/jamf/ajax/import.php at line 58

**Comment by cconard96** [comment]
Try using the 2.0.0 release instead. The master branch is WIP for the next major version.

**Comment by cconard96** [comment]
I should add that you should run the uninstall of the current version of the plugin first, then delete the plugin folder and replace it with the 2.0.0 release. Just changing the files may make the install fail.

**Comment by dnyt84** [comment]
so that is how I started.
I disabled, uninstalled and removed 2.0.1 (master)
and downloaded and unziped 2.0.0 and now I get the following:

2020-08-07 14:47:10 [2@glpi001.xxx.pop]
Undefined function PluginJamfCron::cronimportJamf (for cron)

**Comment by cconard96** [comment]
In the `glpi_crontasks` table, make sure the entries with the itemtype "PluginJamfCron" are deleted. Since those entries were added first, they are taking priority over the correct ones which use "PluginJamfSync" in this version.
There is a bug in GLPI that breaks un-registering automatic actions from plugins. There is a fix, but it missed the 9.5.1 release.

**Comment by dnyt84** [comment]
I uninstalled Jamf Plugin, deleted all Jamf Cron in glpi_crontask and installed Jamf plugin again, again I see nothing.
When i click on Discover I get
This is in the PHP error log:
[2020-08-08 12:29:18] glpiphplog.NOTICE: Toolbox::decrypt() in /var/www/glpi/inc/toolbox.class.php line 311
Use sodiumDecrypt  {"user":"2@glpi001.xxx.pop"} 

The

**Comment by cconard96** [comment]
That warning is expected, this plugin doesn't use the new encryption functions yet.
It sounds like the discover action is running now, but it isn't staging anything for manual import for some reason. Also, you definitely have mobile devices and not just Computers right? The current version only works with mobile devices (iPhones, iPads, Apple TVs).

Here is a patch that adds some debug messages to the php-errors.log file when you try to discover new devices (assuming auto-import is still off and the log level under General > System is set to level 5 Complete).
https://github.com/cconard96/jamf/commit/1684a5d1c49529b737aea1b32545faed9692d21b.patch

Please try making those additions and trying again, then check the php-errors.log file to see if any devices are found and if they get to the "Staging for manual import" stage. Also check for any sql errors in sql-errors.log.

**Comment by dnyt84** [comment]
Ha that would explain things a bit.

We only have computers and no mobile devices in Jamf.

What are the plans to support computers?

Thank you!

Daniel

**Comment by cconard96** [comment]
XD I was worried since it worked in two environments for me. Computer support was originally going to be in this current version but I pushed it back to make sure it was implemented properly while still having GLPI 9.5 support out in a reasonable timeframe.

**Comment by dnyt84** [comment]
If you need a tester for PC support please let me know.
We have about 400 devices in Jamf that I would like to sync into glpi.

We are pre production with GLPI so we can still break things :)

Thank you for taking the time and answering though!

**Comment by cconard96** [comment]
At this time, I'm just trying to modify the current syncing engine to be flexible enough to accommodate both types of devices. Admittedly I was too short-sighted when I created the plugin because my employer only has mobile devices, so Computer support was secondary.
As someone who has computers only, what are your expectations of a plugin integration with GLPI? For mobile devices I added the ability to send mobile commands, is there any need for more MDM features for computers in addition to adding the computer data into GLPI? If so, what are those requirements/expectations?
